tests/test_misc.py

```diff
@@ -30,3 +32,126 @@ def test_digest_file_utf8(self):
 
     def test_digest_file_binary(self):
         self.check_digest(struct.pack('III'.encode('ascii'), 1, 2, 3))
+
+
+class TestUnwrap(unittest.TestCase):
```

Comment:
nit: A fair number of these tests are testing more than one thing. Would it be possible to refactor them to be parameterized?